### PR TITLE
[crystal][laser_geometry] use crystal branch as the upstream.

### PR DESCRIPTION
**Pull request by seanyen**
A recent [change](https://github.com/ros-perception/laser_geometry/pull/46) from `ros2` branch of `laser_geometry` introduces a new dependency `eigen3_cmake_module` which doesn't exist on crystal. If someone builds the crystal from the upstream repository, it will hit build break. So I suggest to swap the source to `crystal` branch instead.

### CODE REVIEW

**Comment by seanyen** [comment]
@sloretz @jonbinney JFYI, I think the `laser_geometry` might need some updates for its source branch `crystal` distro.

**Comment by nuclearsandwich** [comment]
@sloretz this seems like the right thing to do to me. Even if we are going to backport the eigen3_cmake_module package to Crystal and the two laser_geometry changes along with it that should be done via explicit backport.

**Review by nuclearsandwich — APPROVED** [review]
I'd like to get a review from @sloretz as well but this seems like the right change to me.

**Review by sloretz — APPROVED** [review]
This looks right to me too. Introducing an `eigen3_cmake_module` dependency in Crystal was unintentional. I don't have any plans to backport it.

**Comment by jonbinney** [comment]
Sounds good, and yes, we can backport any crucial functionality from the ros2 branch to the crystal branch as needed. @seanyen any PR's in particular you think should be backported?

**Comment by seanyen** [comment]
> @seanyen any PR's in particular you think should be backported?

I think the answer would be related to whether the maintainer of `eigen3_cmake_module` would like to make a release and support it in `crystal`. @sloretz any comments?

**Comment by nuclearsandwich** [comment]
I'm going to merge this change since it reconciles what you'd get building from upstream source repositories with what you'd get building our packages.

I don't think changes related to eigen3_cmake_module need to be backported into Crystal to resolve any issues.

**Pull request merged**
